assets/js/type-defs/registered-payment-method-props.js

```diff
@@ -0,0 +1,206 @@
+ * @typedef {import('@woocommerce/type-defs/contexts').PaymentStatusDispatch} PaymentStatusDispatch
+ * @typedef {import('@woocommerce/type-defs/contexts').ShippingErrorTypes} ShippingErrorTypes
+ * @typedef {import('@woocommerce/type-defs/settings').WooCommerceSiteCurrency} SiteCurrency
+ */
+
+/**
+ * Payment Event Status Action Creators
+ *
+ * @typedef {Object} PaymentStatusActions
+ *
+ * @property {Function} started
+ * @property {Function} processing
+ * @property {Function} completed
+ * @property {Function} error
+ * @property {Function} failed
+ * @property {Function} success
+ */
+
+/**
+ * @typedef {Object} PaymentMethodShipping
+ *
+ * @property {boolean}              required      If shipping is required this
+ *                                                will be true.
+ * @property {CartShippingOption[]} options       Available shipping options.
+ * @property {Function}             selectOptions Used to set the selected
+ *                                                shipping options.
+ * @property {Function}             updateAddress Used to update the shipping
+ *                                                address.
+ * @property {Function}             setStatus     Used to set the current
+ *                                                shipping options status.
+ * @property {Function}             selectStatus  Returns helpers for
+ *                                                determining the current
+ *                                                shipping options status.
+ * @property {string}               status        What the current shipping
+ *                                                options status is.
+ */
+
+/**
+ * @typedef {Object} PaymentMethodEvents
+ *
+ * @property {boolean}  isCalculating      If true, means the cart/checkout is
+ *                                         currently calculating totals.
+ * @property {boolean}  isCheckoutComplete If true, means the checkout process
+ *                                         is complete (useful if the payment
+ *                                         method has something to do after
+ *                                         checkout is complete before setting
+ *                                         payment status to complete. Redirect
+ *                                         doesn't happen until both checkout
+ *                                         and payment status is complete)
+ * @property {PaymentStatusActions} dispatchStatus     Used to dispatch a payment event.
+ * @property {Function} selectStatus       Returns helpers for determining
+ *                                         current payment event status.
+ * @property {string}   status             What the current payment event status
+ *                                         is.
+ */
+
+/**
+ * @typedef CheckoutStatusProps
+ *
+ * @property {boolean} isCalculating If true then totals are being calculated in
```

Comment:
Can some of these status concepts be defined in one place, since they are similar? Examples: `complete`, `calculating`, `processing`

Reply:
My response is the same as [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1929#discussion_r390868760)